packages/did-provider-ebsi/src/ebsi-did-provider.ts

```diff
@@ -0,0 +1,196 @@
+  ICredentialPlugin,
+  IIdentifier,
+  IKey,
+  IKeyManager,
+  IResolver,
+  IService,
+  ManagedKeyInfo,
+} from '@veramo/core'
+import * as jose from 'jose'
+import { AbstractIdentifierProvider } from '@veramo/did-manager'
+import { onboard } from './ebsi-did-onboarding.js'
+import {
+  generateEbsiSubjectIdentifier,
+  generateRandomEbsiSubjectIdentifier,
+  privateKeyJwkToHex,
+} from './ebsi-did-utils.js'
+import { IEbsiCreateIdentifierOptions, IEbsiDidSupportedKeyTypes } from './types/ebsi-provider-types.js'
+import ec from 'elliptic'
+
+type IContext = IAgentContext<IKeyManager & ICredentialPlugin & IResolver>
+
+/**
+ * {@link @veramo/did-manager#DIDManager} identifier provider for `did:ebsi` identifiers
+ *
+ * @beta This API may change without a BREAKING CHANGE notice.
+ */
+export class EbsiDIDProvider extends AbstractIdentifierProvider {
+  private defaultKms: string
+
+  constructor(options: { defaultKms: string }) {
+    super()
+    this.defaultKms = options.defaultKms
+  }
+
+  async createIdentifier(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    { kms, options }: { kms?: string; options?: IEbsiCreateIdentifierOptions },
+    context: IContext,
+  ): Promise<Omit<IIdentifier, 'provider'>> {
+    if (options?.sequence && options?.sequence.length !== 32) {
+      throw new Error('Subject identifier should be 16 bytes (32 characters in hex string, or Uint8Array) long')
+    }
+
+    if (options?.privateKeyHex && !options?.sequence) {
+      throw new Error('Currently, subject identifier sequence should be provided along with a private key')
+    }
+
+    if (options?.privateKeyHex && options?.privateKeyHex?.length !== 64) {
+      throw new Error('Private key should be 32 bytes (64 characters in hex string) long')
+    }
+    const keyType: IEbsiDidSupportedKeyTypes = options?.keyType || 'Secp256k1'
+    if (keyType !== 'Secp256k1') {
+      throw new Error('Currently, only Secp256k1 key type is supported')
+    }
+    let jwkThumbprint: string
+    let privateKeyJwk: jose.JWK
+    let privateKeyHex: string
+    let publicKeyJwk: jose.JWK
+    let subjectIdentifier: string
+    if (options?.privateKeyHex) {
+      // Import existing custom private key along with the subject identifier
+      const secp256k1 = new ec.ec('secp256k1')
+      privateKeyHex = options.privateKeyHex
+      const privateKey = secp256k1.keyFromPrivate(privateKeyHex, 'hex')
+      privateKeyJwk = {
+        kty: 'EC',
+        crv: 'secp256k1',
+        d: privateKey.getPrivate('hex'),
+        x: privateKey.getPublic().getX().toString('hex'),
+        y: privateKey.getPublic().getY().toString('hex'),
+      }
+      publicKeyJwk = { ...privateKeyJwk }
+      delete publicKeyJwk.d
+      jwkThumbprint = await jose.calculateJwkThumbprint(privateKeyJwk)
+      subjectIdentifier = await generateEbsiSubjectIdentifier(options.sequence)
+    } else {
+      // Generate new key pair
+      const keys = await jose.generateKeyPair('ES256K')
```

Comment:
user can pass keyType, options.keyType

https://github.com/uport-project/veramo/blob/next/packages/did-provider-jwk/src/jwk-did-provider.ts#L30

Reply:
and can pass hashType, options.hashType

At the moment, we support only sha256